### ISSUE TITLE
Outline component for website

### ISSUE THREAD

**Issue by mattrosno** [carbon-design-system/carbon-website#1649]
The tutorial has an outline that consists of a numbered list, where each list item has a heading (could be an anchor) and a description.

https://www.carbondesignsystem.com/tutorial/overview/

Right now it looks like:
![image](https://user-images.githubusercontent.com/1691245/58048717-d123fa80-7b10-11e9-8507-c3d7b92326d4.png)

We could have an "outline" component that acts as a navigation control? Maybe something similar to the "What's inside" section in [Learn Storybook](https://www.learnstorybook.com).

**Comment by jeanservaas** [comment]
I did some explorations on this too and I tried to do a mini audit of the areas where we have stepped content. 

@aagonzales and @shixiedesign made a good point, aside from the tiles, all of our other white container components are reserved for highlighting specs (like the typography page) or tables — not for offsite links. 

So for content as short as this, it would be nice to do something just with rules to add a little space.

Option 1:
![image](https://user-images.githubusercontent.com/43144260/58209482-c3a37780-7cac-11e9-94e5-220ab2da9499.png)

Option 2:
![image](https://user-images.githubusercontent.com/43144260/58209001-b20da000-7cab-11e9-9e91-0c373d74f381.png)

Hover (Will these steps eventually be offsite links, like the Learn storybook example?)

![image](https://user-images.githubusercontent.com/43144260/58209597-f2b9e900-7cac-11e9-8086-261d82747464.png)

<br>

Also, somewhat separate issue, but talked to @connor-leech about this, but I also looked at the stepped content on the Getting Started pages and was wondering if we could start to structure it in a clearer way. 

![Getting_started_designers](https://user-images.githubusercontent.com/43144260/58209990-c81c6000-7cad-11e9-8433-cd0d66d6d2cc.png)
